internal/datanode/l0_compactor.go

```diff
@@ -188,65 +190,87 @@ func (t *levelZeroCompactionTask) compact() (*datapb.CompactionPlanResult, error
 	return result, nil
 }
 
-func (t *levelZeroCompactionTask) linearProcess(ctx context.Context, targetSegments []int64, totalDeltalogs map[int64][]string) ([]*datapb.CompactionSegment, error) {
+func (t *levelZeroCompactionTask) linearProcess(ctx context.Context, l1Segments []*datapb.CompactionSegmentBinlogs, totalDeltalogs map[int64][]string) ([]*datapb.CompactionSegment, error) {
 	log := log.Ctx(t.ctx).With(
 		zap.Int64("planID", t.plan.GetPlanID()),
 		zap.String("type", t.plan.GetType().String()),
-		zap.Int("target segment counts", len(targetSegments)),
+		zap.Int("target segment counts", len(l1Segments)),
 	)
+
+	// just for logging
+	l1SegmentIDs := lo.Map(l1Segments, func(segment *datapb.CompactionSegmentBinlogs, _ int) int64 {
+		return segment.GetSegmentID()
+	})
+
 	var (
 		resultSegments  = make(map[int64]*datapb.CompactionSegment)
 		alteredSegments = make(map[int64]*storage.DeleteData)
 	)
+
+	segmentBFs, err := t.loadBF(l1Segments)
+	if err != nil {
+		return nil, err
+	}
 	for segID, deltaLogs := range totalDeltalogs {
 		log := log.With(zap.Int64("levelzero segment", segID))
 
 		log.Info("Linear L0 compaction start processing segment")
 		allIters, err := t.loadDelta(ctx, deltaLogs)
 		if err != nil {
-			log.Warn("Linear L0 compaction loadDelta fail", zap.Int64s("target segments", targetSegments), zap.Error(err))
+			log.Warn("Linear L0 compaction loadDelta fail", zap.Int64s("target segments", l1SegmentIDs), zap.Error(err))
```

Comment:
actually, L0 compaction will target to L1 &L2, any segments not L0, so `L1SegmentIDs` are misleading here. Please change back.

Reply:
updated